spring-rabbit/src/test/java/org/springframework/amqp/rabbit/annotation/EnableRabbitIntegrationTests.java

```diff
@@ -715,6 +716,27 @@ public String handleWithHeadersExchange(String foo) {
 			return foo.toUpperCase();
 		}
 
+		@RabbitListener(bindings = {
+				@QueueBinding(
+					value = @Queue,
+					exchange = @Exchange(value = "auto.internal", autoDelete = "true", internal = "true"),
+					key = "auto.internal.rk")}
+		)
+		public String handleWithInternalExchange(String foo) {
+			return foo.toUpperCase();
+		}
+
+		@RabbitListener(bindings = {
+				@QueueBinding(
+					value = @Queue,
+					exchange = @Exchange(value = "auto.internal", autoDelete = "true",
+								ignoreDeclarationExceptions = "true"),
+					key = "auto.internal.rk")}
+		)
+		public String handleWithInternalExchangeIgnore(String foo) {
+			return foo.toUpperCase();
+		}
```

Comment:
I am just testing that the context loads ok - before the BPP changes, the test fails to load the context due to the mismatched exchange.